### ISSUE TITLE
TensorRT Plugin

### ISSUE THREAD

**Issue by johnnynunez** [levipereira/deepstream-yolov9#1]
I've see your https://github.com/NVIDIA/TensorRT/pull/3859
Is it possible to have on trt10?
I'm working on jetson agx Orin and now is compatible with cuda 12.5, cudnn 9.1.1 and tensorrt 10.0.1.6.
Also, is it compatible with yolov8?

**Comment by levipereira** [comment]
Yes, it can be easily implemented on TRT 10 and for any version of YOLO since v4.
I will try to find some free time and implement it on 8.5 and 10.0

**Comment by levipereira** [comment]
Yes, it can be easily implemented on TRT 10 and for any version of YOLO since v4, because it's same implementation as End2End Efficient NMS but add a new layer det_indices.
I will try to find some free time and implement it on 8.5 and 10.0

**Comment by levipereira** [comment]
https://github.com/NVIDIA/TensorRT/pull/3859#issuecomment-2125899520

**Comment by levipereira** [comment]
@johnnynunez   Check this out.
https://github.com/levipereira/ultralytics -- Added Support for TRT Plugin YoloNMS on Yolov8 for Instance Segmentation and Object Detection
 
I have tested/validated on deepstream with yolov8n --  https://github.com/levipereira/deepstream-yolov9 

 ```python
 from ultralytics import YOLO
 # model = YOLO("yolov8n-seg.pt") 
 model = YOLO("yolov8n.pt") 
 model.export(format="onnx_trt")
 ```

**Comment by johnnynunez** [comment]
awesome! but maybe still I have to do the predict compatible.
These guys did it: https://github.com/nkb-tech/ultralytics

> @johnnynunez Check this out. https://github.com/levipereira/ultralytics -- Added Support for TRT Plugin YoloNMS on Yolov8 for Instance Segmentation and Object Detection
> 
> I have tested/validated on deepstream with yolov8n -- https://github.com/levipereira/deepstream-yolov9
> 
> ```python
> from ultralytics import YOLO
> # model = YOLO("yolov8n-seg.pt") 
> model = YOLO("yolov8n.pt") 
> model.export(format="onnx_trt")
> ```

**Comment by johnnynunez** [comment]
@levipereira also can you create a PR to ultralytics?

**Comment by levipereira** [comment]
@johnnynunez

**Comment by levipereira** [comment]
@johnnynunez 

With Triton Server and Triton Client, we can easily perform inference and evaluation on any YOLO Series model. Check out the evaluation results of YOLOv8 models using YOLO_NMS_TRT at the link below:

[YOLOv8 Evaluation Results](https://github.com/levipereira/triton-server-yolo?tab=readme-ov-file#yolov8)

Implementing inference using the TensorRT API and Custom Plugin within the Ultralytics project involves a significant amount of work. I may consider implementing it in the future.

Using Triton Server, we can build and test any model without additional effort.

For more information, visit:
- [Triton Client for YOLO](https://github.com/levipereira/triton-client-yolo)
- [Triton Server for YOLO](https://github.com/levipereira/triton-server-yolo)

**Comment by levipereira** [comment]
> @levipereira also can you create a PR to ultralytics?

Will implement end2end  with EfficientNMS and YOLO_NMS_TRT  and open a PR.

**Comment by johnnynunez** [comment]
@levipereira do you have lower mAP with efficient_nms in COCO eval?

**Comment by levipereira** [comment]
> @levipereira do you have lower mAP with efficient_nms in COCO eval?

No, I did not get a lower mAP. The results were consistent with the baseline evaluation.

**Comment by levipereira** [comment]
@johnnynunez  https://github.com/levipereira/triton-server-yolo?tab=readme-ov-file#evaluation-test-on-tensorrt
